lua/wikis/commons/Widget/Match/Page/MapVeto.lua

```diff
@@ -0,0 +1,50 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Match/Page/MapVeto
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Widget = Lua.import('Module:Widget')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Div = HtmlWidgets.Div
+local NavigationCard = Lua.import('Module:Widget/MainPage/NavigationCard')
+
+---@class MatchPageMapVetoParameters
+---@field vetoRounds {name: string, link: string, type: 'pick'|'ban'|'decider', round: integer, by: standardOpponent}[]
+
+---@class MatchPageMapVeto: Widget
+---@operator call(MatchPageMapVetoParameters): MatchPageMapVeto
+---@field props MatchPageMapVetoParameters
+local MatchPageMapVeto = Class.new(Widget)
+
+---@return Widget
+function MatchPageMapVeto:render()
+	local formatTitle = function(vetoRound)
+		if vetoRound.type == 'pick' then
+			return 'Pick ' .. vetoRound.by.name
+		elseif vetoRound.type == 'ban' then
+			return 'Ban ' .. vetoRound.by.name
+		elseif vetoRound.type == 'decider' then
+			return 'Decider'
+		end
+	end
+
+	return Div{
+		classes = {'navigation-cards'},
+		children = Array.map(self.props.vetoRounds, function(vetoRound)
+			return NavigationCard{
+				file = vetoRound.name .. ' Map.png',
```

Comment:
is the assumtion correct that the files exist for all maps on all wikis with matchPage?

Reply:
It does on Valorant, which is inscope now. When we get to other wikis that also have map vetos, we need to look at a more scalable solution

Reply:
Can definitely do when there's a need for it, let's just do this for now